trillian/ctfe/handlers.go

```diff
@@ -929,7 +955,27 @@ func buildLogLeafForAddChain(li *logInfo,
 	merkleLeaf ct.MerkleTreeLeaf, chain []*x509.Certificate, isPrecert bool,
 ) (trillian.LogLeaf, error) {
 	raw := extractRawCerts(chain)
-	return util.BuildLogLeaf(li.LogPrefix, merkleLeaf, 0, raw[0], raw[1:], isPrecert)
+	issuanceChain := raw[1:]
+
+	// Trillian gRPC storage backend is enabled and CTFE storage backend is disabled.
+	if li.issuanceChainService.storage == nil {
```

Comment:
Updated to use `li.issuanceChainService.IsCTFEStorageEnabled()`.

I've refactored the code in `addChainInternal` and removed the `buildLogLeafForAddChain`. The issuance chain marshal only performs once now instead of twice.